File: day04/main.py
import hashlib

def Solve(n):
	for x in range(1, 999999999999):
		hex = hashlib.md5(('yzbqklnj%d' % x).encode('utf-8')).hexdigest()
		if hex[0:n] == '0' * n:
			return x

print('Task 1: %d' % Solve(5))
print('Task 2: %d' % Solve(6))

File: day04/main.rb
require 'digest'
Solve = lambda{|l| (0..1000000000).find_index{|k| Digest::MD5.hexdigest("yzbqklnj#{k}")[0...l] == '0'*l} }
puts 'Task 1: %d' % Solve.call(5)
puts 'Task 2: %d' % Solve.call(6)
